cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java

```diff
@@ -214,12 +218,39 @@ protected TimeSeries getTimeseries(String page, int pageSize, String names, Stri
         // Now we're going to call the retrieve_ts_out_tab function to get the data and build an
         // internal table from it so we can manipulate it further
         // This code assumes the database timezone is in UTC (per Oracle recommendation)
-        SQL retrieveSelectData = DSL.sql(
-            "table(cwms_20.cwms_ts.retrieve_ts_out_tab(?,?,cwms_20.cwms_util.to_timestamp(?),cwms_20.cwms_util.to_timestamp(?),"
-           +"'UTC',?,?,?,?,?,?,?,?) ) retrieveTs",
-            tsId,unit, beginTime.toInstant().toEpochMilli(), endTime.toInstant().toEpochMilli(),
-            null,null,null,null,null,null,null,officeId);
-        
+        SQL retrieveSelectData = null;
+
+        Long beginTimeMilli = beginTime.toInstant().toEpochMilli();
+        Long endTimeMilli = endTime.toInstant().toEpochMilli();
+        String trim = null;
+        String startInclusive = null;
+        String endInclusive = null;
+        String previous = null;
+        String next= null;
+        Long versionDateMilli = null;
+        String maxVersion = null;
+
+        if(versionDate != null) {
+            versionDateMilli = versionDate.toInstant().toEpochMilli();
+        } else {
+            maxVersion = "T";
+        }
+
+        // Query based on versionDate or query max aggregate
+        if(versionDate != null) {
+            retrieveSelectData = DSL.sql(
```

Comment:
this should have a comment that to_timestamp will alllow null in next schema release. looked identical on first read.

Reply:
done